src/main/java/com/denizenscript/denizencore/objects/core/ListTag.java

```diff
@@ -1420,22 +1423,20 @@ else if (input.size() > 1) {
                 if (indices.size() > 1) {
                     ListTag results = new ListTag();
                     for (String index : indices) {
-                        int ind = Integer.parseInt(index);
-                        if (ind > 0 && ind <= object.size()) {
-                            results.add(object.get(ind - 1));
+                        int ind = CoreUtilities.parseIndex(object.objectForms, index);
+                        if (ind >= 0) {
+                            results.add(object.get(ind));
+                        } else {
+                            attribute.echoError("Invalid index '" + index + "'.");
                         }
                     }
                     return results;
                 }
                 if (indices.size() > 0) {
-                    int index = Integer.parseInt(indices.get(0)) - 1;
-                    if (index >= object.size()) {
-                        attribute.echoError("Invalid list.get index '" + (index + 1) + "' ... list is only " + object.size() + " long.");
-                        return null;
-                    }
+                    int index = CoreUtilities.parseIndex(object.objectForms, indices.get(0));
                     if (index < 0) {
-                        attribute.echoError("Invalid list.get index '" + (index + 1) + "' ... must be at least 1.");
-                        index = 0;
+                        attribute.echoError("Invalid list.get index '" + (index + 1));

```

Comment:
chopped  this error message weird

Reply:
Chopped it since it can be negative now, really only handles the 0 case there. I'll update it to be specific to that